### PR TITLE
Allow to split bulk update pay items into smaller requests

### PR DESCRIPTION
**Pull request by mmeeoorroo**
# Trello card (URL):
https://trello.com/c/id0Hdhwo/715-get-requesttimeout-when-pull-pay-items-and-cause-tech-difficulties-error

# Summary
Split bulk-update (for pay items) into smaller requests

# Changes proposed in this pull request:
- Allow to customize bulk-size for pay cats, leave cats, work types, locations and pay schedule
- Default bulk size = 100

Can @(reviewer_handle) have a look at this PR. Thanks!

### CODE REVIEW

File: lib/keypay/sync/location.rb [hunk 1]
```diff
@@ -20,27 +20,28 @@ def self.pull_all(eh_context, kp_context, force = false)
...
-          pull_child_locations(kp_locations, child_locations, eh_context)
+          push_locations_to_eh(eh_context, child_locations, bulk_size: bulk_size)
+          pull_child_locations(kp_locations, child_locations, eh_context, bulk_size: bulk_size)
         end
       end
 
-      def self.push_locations_to_eh(eh_context, kp_locations)
-        kp_locations.each_slice((ENV['LOCATIONS_PER_REQUEST']|| 100).to_i) do |grouped_locations|
+      def self.push_locations_to_eh(eh_context, kp_locations, bulk_size: nil)
+        bs = bulk_size.to_i > 0 ? bulk_size : 100
```
Comment (khoido) [inline lib/keypay/sync/location.rb]: should it be `bs = bulk_size.to_i > 0 ? bulk_size : kp_locations .length` ?

**Comment by khoido** [comment]
@mmeeoorroo you should also add tests for bulk logic

File: lib/keypay/sync/location.rb (hunk 1 above)
Comment (mmeeoorroo) [reply lib/keypay/sync/location.rb]: yes you're right. Thanks

File: lib/keypay/sync/location.rb [hunk 2]
```diff
@@ -20,27 +20,28 @@ def self.pull_all(eh_context, kp_context, force = false)
...
-          pull_child_locations(kp_locations, child_locations, eh_context)
+          push_locations_to_eh(eh_context, child_locations, bulk_size: bulk_size)
+          pull_child_locations(kp_locations, child_locations, eh_context, bulk_size: bulk_size)
         end
       end
 
-      def self.push_locations_to_eh(eh_context, kp_locations)
-        kp_locations.each_slice((ENV['LOCATIONS_PER_REQUEST']|| 100).to_i) do |grouped_locations|
+      def self.push_locations_to_eh(eh_context, kp_locations, bulk_size: nil)
+        bs = bulk_size.to_i > 0 ? bulk_size : kp_locations.length
```
Comment (thukim) [inline lib/keypay/sync/location.rb]: bulk_size.to_i otherwise you will get `no implicit conversion of String into Integer` when its a string

Comment (mmeeoorroo) [reply lib/keypay/sync/location.rb]: :facepalm: I forgot it

Comment (mmeeoorroo) [reply lib/keypay/sync/location.rb]: Here it is @thukim https://github.com/Thinkei/keypay-integration/pull/60

**Pull request merged**